scripts/analyze_benchmarks/statistics.ts

```diff
@@ -0,0 +1,133 @@
+import { BenchmarkEntry } from './extractor';
+
+export type Statistics = {
+    count: number;
+    mean: number;
+    median: number;
+    standardDeviation: number;
+    min: number;
+    max: number;
+    baselineWeightedEfficiencyScore: number;
+};
+
+const MAX_VALID_INSTRUCTIONS = 40_000_000_000;
+const EFFICIENCY_WEIGHTS = {
+    min: 0.6,
+    median: 0.3,
+    mean: 0.1
+} as const;
+
+/**
+ * Calculates statistics for a set of benchmark entries
+ * @param entries Array of benchmark entries to analyze
+ * @returns Statistical analysis of the benchmark data
+ */
+export function calculateVersionStatistics(
+    entries: BenchmarkEntry[]
+): Statistics {
+    const instructions = entries.map((entry) =>
+        Number(entry.instructions.__bigint__)
+    );
+
+    return calculateStatistics(instructions);
+}
+
+/**
+ * Calculates statistical measures for an array of instruction counts
+ * @param instructions Array of instruction counts
+ * @returns Statistical analysis including mean, median, standard deviation, etc.
+ * @throws Error if input array is empty or contains no valid instructions
+ */
+function calculateStatistics(instructions: number[]): Statistics {
+    if (instructions.length === 0) {
+        throw new Error('Cannot calculate statistics for empty array');
+    }
+
+    // Filter out instructions > 40 billion
+    const filteredInstructions = instructions.filter(
+        (val) => val <= MAX_VALID_INSTRUCTIONS
+    );
+
+    if (filteredInstructions.length === 0) {
+        throw new Error(
+            'No valid instructions after filtering (all values > 40 billion)'
+        );
+    }
+
+    const sorted = [...filteredInstructions].sort((a, b) => a - b);
```

Comment:
Lambda necessary?

Reply:
yes or else it will sort it alphabetically instead of numerically